### PR TITLE
Opening searcher with source node

### LINKED ISSUE #8065: Opening searcher with source node
To be done after #8064

If Component Browser is opened while another node is selected, or it was opened by dragging out an edge (if #8054 will be implemented), the CB input should be prepopulated with source node identifier and dot. If the user types an operator right after opening, the dot should be replaced with space.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Implements #8065.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/ComponentBrowser.vue [hunk 1]
```diff
@@ -27,18 +28,38 @@ const props = defineProps<{
...
   finished: [selectedExpression: string]
 }>()
 
+const initialContent = computed(() => {
+  if (props.sourceNode == null) return props.initialContent
+  const sourceNode = props.sourceNode
+  const sourceNodeName = useGraphStore().getNodeBinding(sourceNode)
+  const sourceNodeNameWithDot = sourceNodeName ? sourceNodeName + '.' : ''
+  return sourceNodeNameWithDot + props.initialContent
+})
```
Comment (farmaazon) [inline app/gui2/src/components/ComponentBrowser.vue]: I think this and `initialCaret` does not need to be `computed`, functions should be enough: we want to get this information only at specific event of opening component browser.

Also, we do not actually expect to have both source node _and_ initial content (you create a node with source node known or edit an existing node without any source).

File: app/gui2/src/components/ComponentBrowser.vue [hunk 2]
```diff
@@ -27,18 +28,38 @@ const props = defineProps<{
...
+
+const initialCaret = computed((): ContentRange => {
+  if (props.sourceNode == null) return props.initialCaretPosition
+  const sourceNode = props.sourceNode
+  const sourceNodeName = useGraphStore().getNodeBinding(sourceNode)
```
Comment (farmaazon) [inline app/gui2/src/components/ComponentBrowser.vue]: The store should be obtained in modules scope, like `projectStore` or `suggestionDbStore` below.

**Review by farmaazon — APPROVED** [review]
I guess it has a bug: when editing existing node when another node is selected, we add its name with dot at the beginning, what is not expected. I guess it's a trivial fix, so I don't need to check code again, I think.

Also, I miss the following section from the issue's description:

> If the user types an operator right after opening, the dot should be replaced with space.

But that could be added in a separate PR (or extracted to another task, it's not a priority now).

**Comment by MichaelMauderer** [comment]
> I guess it has a bug: when editing existing node when another node is selected, we add its name with dot at the beginning, what is not expected. I guess it's a trivial fix, so I don't need to check code again, I think.

Fixed 


> Also, I miss the following section from the issue's description:
> 
> > If the user types an operator right after opening, the dot should be replaced with space.
> 
> But that could be added in a separate PR (or extracted to another task, it's not a priority now).

Let's do it separately, as it seems somewhat independent of this functionality.

**Review by Frizi — APPROVED** [review]

**Pull request merged**